### PR TITLE
DO NOT MERGE added in failing when demonstrate list types the aea is unable to parse

### PR DESCRIPTION
**Pull request by 8ball030**
## Proposed changes

this pr adds in a test for non json dumped list types to apply environment variables.

```
value = "['a']", type_str = 'list'

    def convert_value_str_to_type(value: str, type_str: str) -> JSON_TYPES:
        """Convert value by type name to native python type."""
        try:
            type_ = FROM_STRING_TO_TYPE[type_str]
            if type_ == bool:
                return value not in FALSE_EQUIVALENTS
            if type_ is None or value in NULL_EQUIVALENTS:
                return None
            if type_ in (dict, list):
                return json.loads(value)
            return type_(value)
        except (ValueError, json.decoder.JSONDecodeError):  # pragma: no cover
>           raise ValueError(f"Cannot convert string `{value}` to type `{type_.__name__}`")  # type: ignore
E           ValueError: Cannot convert string `['a']` to type `list`

```

```
pytest tests/test_helpers/test_env_vars.py 
```

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: tests/test_helpers/test_env_vars.py [hunk 1]
```diff
@@ -78,3 +78,12 @@ def test_convert_value_str_to_type():
     assert convert_value_str_to_type("none", "str") is None
     assert convert_value_str_to_type("null", "str") is None
     assert convert_value_str_to_type("None", "str") is None
+
+def test_convert_list_string_to_type():
+    """Test convert_string lists to list."""
+    assert convert_value_str_to_type("['a']", "list") == ['a']
```
Comment (8ball030) [inline tests/test_helpers/test_env_vars.py:84]: This fails. It is also the output of calling string on a list in python

```
In [15]: str(["a"])
Out[15]: "['a']"
```

File: tests/test_helpers/test_env_vars.py [hunk 2]
```diff
@@ -78,3 +78,10 @@ def test_convert_value_str_to_type():
     assert convert_value_str_to_type("none", "str") is None
     assert convert_value_str_to_type("null", "str") is None
     assert convert_value_str_to_type("None", "str") is None
+
+
+def test_convert_list_string_to_type():
+    """Test convert_string lists to list."""
+    assert convert_value_str_to_type(str(['a']), "list") == ['a']
```
Comment (8ball030) [inline tests/test_helpers/test_env_vars.py:85]: I dont think this test should fail.

Comment (DavidMinarsch) [reply tests/test_helpers/test_env_vars.py:85]:
```
>>> import json
>>> str(['a'])
"['a']"
>>> json.dumps(['a'])
'["a"]'
>>> 
```
The assumption is that the str is json encoded for list and dict (as we can see from the implementation)

Comment (8ball030) [reply tests/test_helpers/test_env_vars.py:85]: Then i can call json.dumps on the base_deployment yes?

Comment (DavidMinarsch) [reply tests/test_helpers/test_env_vars.py:85]: Nothing needs calling anywhere. The yaml needs to specify the correctly formatted values.

**Pull request closed without merging**